internal/libwebp/src/webp.c

```diff
@@ -87,6 +87,97 @@ uint8_t* webpDecodeRGBA(
 	return WebPDecodeRGBA(data, data_size, width, height);
 }
 
+int webpDecodeGrayToSize(const uint8_t* data, size_t data_size,
+	int width, int height,
+	uint8_t* gray, size_t gray_size
+) {
+	if (gray_size < width * height) {
+		return -1;
+	}
+
+	WebPDecoderConfig config;
+	if(!WebPInitDecoderConfig(&config)) {
+		return -1;
+	}
+
+	config.options.bypass_filtering = 1;
+	config.options.no_fancy_upsampling = 1;
+	config.options.use_scaling = 1;
+	config.options.scaled_width = width;
+	config.options.scaled_height = height;
+	config.output.colorspace = MODE_YUV;
+
+	int status = WebPDecode(data, data_size, &config);
+	if(status != VP8_STATUS_OK) {
+		return status;
+	}
+
+	size_t stride = config.output.u.YUVA.y_stride;
+	uint8_t* src = config.output.u.YUVA.y;
+	uint8_t* dst = gray;
+	int i;
+
+	if(stride == width && config.output.u.YUVA.y_size == gray_size) {
+		memcpy(dst, src, gray_size);
+	}
+	else {
+		for(i = 0; i < height; ++i) {
+			memmove(dst, src, width);
+			src += stride;
+			dst += width;
```

Comment:
`dst += width`, use `dstStride` or name `widthStep` is better:
`dst += widthStep`